Consider musl provided built-in defines

Helps compile libtirpc with musl
bits/endian.h is not supposed to be included directly

Upstream-Status: Pending
Signed-off-by: Khem Raj <raj.khem@gmail.com>

--- a/tirpc/rpc/types.h
+++ b/tirpc/rpc/types.h
@@ -66,7 +66,7 @@ typedef   int32_t rpc_inline_t;
 #define mem_free(ptr, bsize)	free(ptr)
 
 
-#if defined __APPLE_CC__ || defined __FreeBSD__
+#if defined __APPLE_CC__ || defined __FreeBSD__ || !defined(__GLIBC__)
 # define __u_char_defined
 # define __daddr_t_defined
 #endif
--- a/src/xdr_float.c
+++ b/src/xdr_float.c
@@ -83,7 +83,7 @@ static struct sgl_limits {
 };
 #else
 
-#include <bits/endian.h>
+#include <endian.h>
 #define IEEEFP
 
 #endif /* vax */
